templates/vue/src/components/TapestryMain/RootNodeButton/index.vue

```diff
@@ -143,42 +217,96 @@ export default {
     },
     importTapestryFromZip(zipFile) {
-      this.error = ""
+      this.error = {}
       this.isImporting = true
+      this.importStatusMessage = "Getting ready to import"
+
       client
-        .importTapestryFromZip(zipFile)
+        .clearImportStatus()
+        .catch(err => {
+          this.error = err.response.data
+        })
+        .finally(() => {
+          this.updateImportStatus()
+        })
+
+      this.cancelTokenSource = axios.CancelToken.source()
+
+      client
+        .importTapestryFromZip(zipFile, this.cancelTokenSource)
         .then(response => {
+          let shouldRebuild = false
           if (response) {
             this.changes.permissions = new Set(response.changes.permissions)
             this.changes.noChange = response.changes.noChange
             this.warnings = response.warnings
             this.exportWarnings = response.exportWarnings
 
-            return response.rebuildH5PCache
+            shouldRebuild = response.rebuildH5PCache
           } else {
-            return false
+            this.error = {
+              message: "An error occurred while uploading file.",
+            }
+            shouldRebuild = false
           }
+          return shouldRebuild
         })
         .catch(err => {
-          this.error = err.response.data
+          if (err.response) {
+            this.error = err.response.data
+          } else {
+            this.error = {
+              message: err.message,
+            }
+          }
         })
         .then(shouldRebuild => {
           if (shouldRebuild) {
             // The h5pMeta.details field is not generated for imported H5Ps
             // If any H5Ps were added during import, we need to rebuild the H5P cache
+            this.importStatusMessage =
+              "Rebuilding H5P cache... Please do not close this tab."
             return WordpressApi.rebuildAllH5PCache()
           }
         })
         .catch(err => {
           this.error = err
         })
         .finally(() => {
-          if (!this.error) {
+          if (!this.error?.message) {
             this.$bvModal.show("import-changelog")
           }
+          this.importStatusMessage = ""
+          this.showImportProgress = false
           this.isImporting = false
         })
     },
+    cancelUpload() {
+      this.cancelTokenSource.cancel("You cancelled the upload.")
+    },
+    updateImportStatus() {
+      client
+        .getImportStatus()
+        .then(status => {
+          if (!this.isUploading && status.inProgress && status.message) {
+            this.importStatusMessage =
+              status.message + "... Please do not close this tab."
+          }
+        })
+        .catch(err => {
+          this.addApiError(err)
+        })
+        .finally(() => {
+          if (this.isImporting) {
+            setTimeout(
+              () => {
+                this.updateImportStatus()
+              },
+              this.isUploading ? 2000 : 50
```

Comment:
Not sure if the 50 here is intended or not. When testing locally, 50ms is great and the import status updates promptly. However, I'm not sure if we want this short of a timeout between requests to the server.

Reply:
After discussion, we have decided to increase the timeout between requests to 2000ms, since 50ms is indeed too small and would cause lots of requests being sent to the server.